### PR TITLE
Add binding event

### PR DESCRIPTION
**Pull request by pigletfly**
Signed-off-by: pigletfly <wangbing.adam@gmail.com>

**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes https://github.com/karmada-io/karmada/issues/398

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by RainbowMango** [comment]
I can't wait to try it, thanks.

**Comment by pigletfly** [comment]
add `NewResourceBindingController ` and  `NewClusterResourceBindingController` to create Controller instance

**Comment by RainbowMango** [comment]
> add `NewResourceBindingController ` and `NewClusterResourceBindingController` to create Controller instance

Why? I'm afraid of too many parameters.

File: pkg/controllers/binding/binding_controller.go [hunk 1]
```diff
@@ -61,7 +81,10 @@ func (c *ResourceBindingController) Reconcile(ctx context.Context, req controlle
 
 	isReady := helper.IsBindingReady(binding.Spec.Clusters)
 	if !isReady {
-		klog.Infof("ResourceBinding(%s/%s) is not ready to sync", binding.GetNamespace(), binding.GetName())
+		msg := fmt.Sprintf("ResourceBinding %s is not ready.", req.NamespacedName.String())
+		klog.Infof(msg)
+		c.recorder.Event(binding, corev1.EventTypeWarning, "ResourceBindingNotReady", msg)
+
```
Comment (RainbowMango) [inline pkg/controllers/binding/binding_controller.go:87]:
```
Events:
  Type     Reason                   Age                 From                Message
  ----     ------                   ----                ----                -------
  Warning  ResourceBindingNotReady  44s                 binding-controller  ResourceBinding default/nginx-deployment is not ready.
```

This basically adds the `warning` event to `all` ResourceBinding objects. Are you sure you want to do this?

**Comment by pigletfly** [comment]
Variables in Controller struct are used only internally, and also follows kube-conroller-manager style.

**Comment by RainbowMango** [comment]
> Variables in Controller struct are used only internally, and also follows kube-conroller-manager style.

**Comment by RainbowMango** [comment]
I recommend letting this PR focus on the event thing, not the code style.

If we decide to change the code style, then we should change all controllers.

**Comment by RainbowMango** [comment]
Hi @pigletfly 
I propose some similar at https://github.com/karmada-io/karmada/issues/398#issuecomment-925652756.

**Comment by pigletfly** [comment]
> I recommend letting this PR focus on the event thing, not the code style.
> 
> If we decide to change the code style, then we should change all controllers.

OK

**Comment by RainbowMango** [comment]
After discussed with @pigletfly locally. @XiShanYongYe-Chang will help move forward based on this patch.

**Pull request closed without merging**